strawberry/experimental/pydantic/object_type.py

```diff
@@ -43,17 +51,36 @@
 from strawberry.types.type_resolver import _get_fields
 from strawberry.types.types import TypeDefinition
 
-from .exceptions import MissingFieldsListError, UnregisteredTypeException
+from .exceptions import MissingFieldsListError
+
+
+try:
+    from typing import ForwardRef  # type: ignore
+except ImportError:  # pragma: no cover
+    # ForwardRef is private in python 3.6 and 3.7
+    from typing import _ForwardRef as ForwardRef  # type: ignore
+
+from .lazy_types import LazyForwardRefType, LazyModelType
 
 
-def replace_pydantic_types(type_: Any, is_input: bool):
+def replace_pydantic_types(
+    type_: Any, is_input: bool, model: Type[BaseModel], name: str
+):
+    if isinstance(type_, ForwardRef):
+        return LazyForwardRefType[model, name]  # type: ignore
+
+    if is_ormar_model(model):
+        return replace_ormar_types(type_, model, name)
+
     origin = getattr(type_, "__origin__", None)
     if origin is Literal:
         # Literal does not have types in its __args__ so we return early
         return type_
     if hasattr(type_, "__args__"):
         replaced_type = type_.copy_with(
-            tuple(replace_pydantic_types(t, is_input) for t in type_.__args__)
+            tuple(
+                replace_pydantic_types(t, is_input, model, name) for t in type_.__args__
+            )
         )
```

Comment:
Hi @gazorby   
  
Many thanks for your work on this :-)  
  
I'm testing the PR against a set of nested Pydantic Models. In this section, I'm running into an issue with pydantic fields using a list like `orders: Optional[list[Order]]` and using a dict like `name_dir: dict[str, Any]` where it then throws `AttributeError: type object 'dict' has no attribute 'copy_with'` or `AttributeError: type object 'list' has no attribute 'copy_with'`

Reply:
Hi @alexhafner 

Thanks for your reports, I think we don't support builtin `list` yet, only `List` from the typing module. I tested with the following nested structure and it works:

```python
    class HobbyModel(pydantic.BaseModel):
        name: str

    class UserModel(pydantic.BaseModel):
        age: int
        hobbies: Optional[List[HobbyModel]]

    @strawberry.experimental.pydantic.type(HobbyModel, fields=["name"])
    class Hobby:
        pass

    @strawberry.experimental.pydantic.type(UserModel, fields=["age", "hobbies"])
    class User:
        pass
```

Reply:
Thanks for that @gazorby - I have changed the underlying pydantic models using list[] to List[] and it works fine. dict[...] or Dict[...] doesn't seem to be supported at all.

Reply:
some of the checks in `if issubclass(type_, BaseModel):` fail with `issubclass() arg 1 must be a class`. For example if you try to use `Dict[str, str]` you get `TypeError: Foo fields cannot be resolved. Unexpected type 'typing.Dict[str, str]'`. With `Dict[str, Any]` you end up with `TypeError: issubclass() arg 1 must be a class`. Once you add an additional class check like `if inspect.isclass(type_) and issubclass(type_, BaseModel):` then the expected error is returned `TypeError: Foo fields cannot be resolved. Unexpected type 'typing.Dict[str, typing.Any]'`

Reply:
Does strawberry currently support `typing` containers, e.g. `typing.Set`? I see errors like 

```
TypeError: Unexpected type 'typing.Set[pydantic.networks.EmailStr]'
```

Reply:
GraphQL doesn't have a Set container type, only List. You'll have to convert it to list before you instantiate the strawberry ObjectType

Reply:
Oh, gotcha, thanks for the pointer and apologies for hijacking the PR here 🙇

For anyone landing on this page, here are [the docs](https://strawberry.rocks/docs/integrations/pydantic#custom-conversion-logic) and my code to solve this:

```python
    @staticmethod
    def from_pydantic(
        instance: MyPydanticModel,
        extra: Optional[Dict[str, Any]] = None,
    ) -> "MyModel":
        data = instance.dict()
        data['emails'] = list(data.get('emails', []))
        return MyModel(**data)

    def to_pydantic(self) -> MyPydanticModel:
        data = dataclasses.asdict(self)
        data['emails'] = {EmailStr(email) for email in data.get('emails', [])}
        return MyPydanticModel(**data)
```